nodebuilder/da/service.go

```diff
@@ -28,12 +29,32 @@ var log = logging.Logger("go-da")
 const heightLen = 8
 
 type Service struct {
-	blobServ nodeblob.Module
+	blobServ     nodeblob.Module
+	headerGetter func(context.Context, uint64) (*header.ExtendedHeader, error)
 }
 
-func NewService(blobMod nodeblob.Module) *Service {
+// SubmitOptions defines options for blob submission using SubmitWithOptions.
+//
+// SubmitWithOptions expects JSON serialized version of this struct; all fields are optional.
+type SubmitOptions struct {
+	// KeyName is the name of key from the keystore used to sign transactions.
+	KeyName string
+
+	// SignerAddress is the address that signs the transaction.
+	// This address must be stored locally in the key store.
+	SignerAddress string
+
+	// FeeGranterAddress specifies the account that will pay for the transaction fee.
+	FeeGranterAddress string
+}
```

Comment:
Pls add json tags, so we can use them in API docs and other langs can implement options support

Reply:
Added JSON tags, and used `snake_case` for consistency.